packages/tools/client-debugger/client-debugger-view/src/WindowMessageRelay.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import {
+	IDebuggerMessage,
+	ISourcedDebuggerMessage,
+	IMessageRelay,
+	IMessageRelayEvents,
+	isDebuggerMessage,
+	debuggerMessageSource,
+} from "@fluid-tools/client-debugger";
+
+/**
+ * Message relay used by a debugger view rendered in the same page as the application to communicate with the
+ * {@link @fluid-tools/client-debuger#DebuggerRegistry}.
+ *
+ * @remarks
+ *
+ * While a debugger view rendered in the same page as the application could technically communicate with the
+ * {@link @fluid-tools/client-debuger#DebuggerRegistry} directly, we put this abstraction in the middle to match the
+ * way that a debugger view rendered outside the context of the application (e.g. the browser's DevTools panel) has
+ * to communicate with the debugger registry.
+ * This ensures that we don't "abuse" the power of local interaction to do things that might not be possible (or need
+ * to be done differently) with a message passing mechanism that crosses the boundary of the window.
+ */
+export class WindowMessageRelay
+	extends TypedEventEmitter<IMessageRelayEvents>
+	implements IMessageRelay
+{
+	public constructor(private readonly messageSource: string) {
```

Comment:
Might be worth adding a brief comment to `messageSource` linking to ISourcedMessage for context.